fe1-web/src/features/evoting/reducer/ElectionReducer.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
+
+import { Hash } from 'core/objects';
+
+import { Election, ElectionState } from '../objects';
+
+/**
+ * Reducer & associated function implementation to store all known elections
+ */
+
+export interface ElectionReducerState {
+  byId: Record<string, ElectionState>;
+  allIds: string[];
+}
+
+const initialState: ElectionReducerState = {
+  byId: {},
+  allIds: [],
+};
```

Comment:
Is this related to any LAO or is it completely generic and stores all elections?
The same questions applies for each <Event>Reducer.

Reply:
Yes this is supposed to be generic and store all elections. The ids of the different elections, roll calls, etc. are obtained by computing a hash function on some constants, the laoId and some other properties. Thus as long as the hash function is collision resistant (which it should be, otherwise we have problems with the security-related parts of the protocol), the ids will be unique. And as you said, if you would like to obtain all elections for a given lao, you can use the events reducer to obtain the set of ids and then load them from the election reducer. As far as I remember, this is only done for roll calls in the wallet and in all other placed we do not need a list of a given event type for a given lao id.

Reply:
Okay, thanks for the explanation. This implementation seems fine to me.